components/brave_news/rust/lib.rs

```diff
@@ -26,10 +27,55 @@ mod ffi {
     }
 
     extern "Rust" {
+        fn strip_html(subject: &str) -> String;
         fn parse_feed_bytes(source: &[u8], output: &mut FeedData) -> bool;
     }
 }
 
+// Note: This function isn't likely to be perfect, but it does guarantee that
+// there will be no HTML tags in the output (as it strips both '<' and '>' from
+// from the output).
+fn strip_html(subject: &str) -> String {
+    let mut output = String::with_capacity(subject.len());
+    let mut depth_tag = 0;
+    let mut depth_comment = 0;
+
+    for (i, c) in subject.chars().enumerate() {
+        if c == '<' {
+            let is_comment = subject.chars().skip(i).take(4).collect::<String>() == "<!--";
```

Comment:
Isn't re-scanning with a new iterator like this also O(N^2)? I wonder if you can do something with `while let Some(c) = subject_chars.next()` and `Chars::as_str()` to keep the starting point local? Still extra work because you need some way to bounds-check your lookahead (e.g. `.as_str().get(..3)` maybe?) but could still be more readable than scanning the bytes yourself.

Reply:
You're right here - it's still `O(n^2)`, I'll look into just indexing for substrings. That said, it's multiple orders of magnitude faster. Maybe grapheme calculations are expensive?

Reply:
Swapped to just using the iterator.